activesupport/lib/active_support/testing/time_helpers.rb

```diff
@@ -1,32 +1,38 @@
+require 'concurrent'
+
 module ActiveSupport
   module Testing
     class SimpleStubs # :nodoc:
       Stub = Struct.new(:object, :method_name, :original_method)
 
       def initialize
-        @stubs = {}
+        @stubs = Concurrent::Map.new{ |h,k| h[k] = {} }
       end
 
       def stub_object(object, method_name, return_value)
-        key = [object.object_id, method_name]
-
-        if stub = @stubs[key]
+        if stub = @stubs[object.object_id][method_name]
```

Comment:
We could rename `stubbed?` to `stubbing` or something else that's less predicate-ish. That way we can reuse it here 😁